src/conv.rs

```diff
@@ -345,9 +354,16 @@ unsafe impl<'a, V: FromValue<'a>> FromValue<'a> for Vec<V> {
     fn from_value(v: Value) -> Vec<V> {
         unsafe {
             let len = crate::sys::caml_array_length(v.raw().0);
+            let is_double = sys::caml_is_double_array(v.raw().0) == 1 && sys::FLAT_FLOAT_ARRAY;
             let mut dst = Vec::with_capacity(len);
+            let mut tmp = Value::f64(0.0);
             for i in 0..len {
-                dst.push(V::from_value(Value::new(*crate::sys::field(v.raw().0, i))))
+                if is_double {
```

Comment:
Maybe it's a nitpick but wouldn't this `if` be better outside the loop?

Reply:
Good idea, I will move that if statement outside the loop!